matrix-sdk-android/src/test/java/org/matrix/android/sdk/test/fakes/FakeWorkManagerProvider.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2022 The Matrix.org Foundation C.I.C.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.matrix.android.sdk.test.fakes
+
+import androidx.work.OneTimeWorkRequest
+import androidx.work.WorkManager
+import io.mockk.every
+import io.mockk.mockk
+import org.matrix.android.sdk.internal.di.WorkManagerProvider
+
+internal class FakeWorkManagerProvider {
+
+    val instance = mockk<WorkManagerProvider>()
+
+    init {
+        val workManager = mockk<WorkManager>()
```

Comment:
I'm wondering if we should be explicitly expecting the tests to mock `enqueueUniqueWork` `cancelUniqueWork` rather than applying it by default, we can make the test cases more resilient by also verifying against the arguments  

```kotlin
class FakeWorkManagerProvider(val fakeWorkManager = FakeWorkManager()) {
    val instance = mockk<WorkManagerProvider>().also {
        every { workManager } returns fakeWorkManager.instance
    }
}

class FakeWorkManager {
    val instance = mockk<WorkManager>()

    fun expectCancelUniqueWork() {
        every { instance.cancelUniqueWork(any()) } returns mockk()
    }

   fun verifyCancelUniqueWork(name: String) {
     verify { instance.cancelUniqueWork(name) }
   }

}
```

Reply:
Yes you are right. I will make an update.

Reply:
Unpdated here: [454209a](https://github.com/vector-im/element-android/pull/6267/commits/454209a826be83a8b39d7b843624dbe184eb78e3)